src/scripts/github_activity_detector.py

```diff
@@ -0,0 +1,56 @@
+import os
+import datetime
+import time
+from yeelight import Bulb
+from github import Github
+from dotenv import load_dotenv
+
+load_dotenv()
+
+yeelight_ip = os.environ["BULB_IP"]
+bulb = Bulb(yeelight_ip)
+token = os.environ["PERSONAL_ACCESS_TOKEN"]
+
+g = Github(token)
+
+user = g.get_user("Samueljbk")
+
+
+def set_light_colour(colour):
+    if colour == "red":
+        bulb.set_rgb(255, 0, 0)
+    elif colour == "green":
+        bulb.set_rgb(0, 255, 0)
+
+
+def check_commit_status():
+    all_events = list(user.get_events())
+    push_events = [e for e in all_events if e.type == "PushEvent"]
+    latest_event = max(push_events, key=lambda e: e.created_at)
+    latest_event_time = latest_event.created_at
+
+    if latest_event_time.date() == datetime.date.today():
+        print("Pushed Today")
+        set_light_colour("green")
+    else:
+        print("No Push Today")
+        set_light_colour("red")
+
+
+check_commit_status()
+
+while True:
+    current_time = datetime.datetime.now()
+    if current_time.hour == 0 and current_time.minute == 0:
+        set_light_colour("red")
+        time.sleep(60)
+    else:
```

Comment:
I don't think this is needed as the resulting behaviour would probably be the same anyway

Reply:
Done